libinteract/libinteract.py

```diff
@@ -866,156 +867,140 @@ def calc_cg_fullmatrix(identifiers, idxs, percmat, perco):
     # return the full matrix (square matrix)
     return fullmatrix
 
-def filter_by_chain(chain1, chain2, sec_res, array, array_T):
-    """Takes in a table with 2 residue columns and filters it to 
-    return only rows where the first column contains chain1 and the
-    second column contains chain2. Returns filtered table.
+def filter_by_chain(chain1, chain2, sec_res, table):
+    """Takes in a table, its transpose, two chain names and the column
+    number of the second chain. Filters the table to only keep rows 
+    where the first column contains chain1 and the second column contains
+    chain2. Returns filtered table.
     """
 
+    # Get the rows and columns from the table
+    rows = table
+    cols = table.T
     # Filter and create filtered array
-    logical_vector = np.logical_and(chain1 == array_T[0], \
-                                    chain2 == array_T[sec_res])
-    filtered_array = array[logical_vector]
-    output = None
+    logical_vector = np.logical_and(chain1 == cols[0],
+                                    chain2 == cols[sec_res])
+    filtered_rows = rows[logical_vector]
     # Warn if no contacts found
-    if filtered_array.shape[0] == 0:
+    if filtered_rows.shape[0] == 0:
         if chain1 == chain2:
-            warnstr = f"No intrachain contacts found in chain {chain1}"
-            log.warning(warnstr)
+            log.warning(f"No intrachain contacts found in chain {chain1}")
         else:
             warnstr = f"No interchain contacts found between chains " \
                       f"{chain1} and {chain2}"
             log.warning(warnstr)
+        return None
     else:
-        output = filtered_array
-    return output
+        return filtered_rows
 
 
-def create_table_list(contact_list, hb=False):
-    """Takes in a list of tuples and returns a list of arrays where
-    the first array contains all contacts followed by intrachain 
-    contants split by chain (if present) and interchain contacts split 
-    by chain (if present).
+def create_table_dict(table, hb=False):
```

Comment:
can you just check the size of the array instead of having an extra argument?

this is also for the other function with the `hb` argument

Reply:
I got rid of the `hb` argument but I tried using a function to find the positions instead of hard coding them based on shape. I'm not sure if the position function should actually be two functions or just part of the `create_table_dict`  function. I added an assert statement to make sure I find the correct number of positions, but I'm not sure if that should be there at all or if it should be a `log.error`